controllers/vdoconfig_controller.go

```diff
@@ -901,44 +1000,177 @@ func (r *VDOConfigReconciler) reconcileCSISecret(ctx vdocontext.VDOContext, conf
 	return config, nil
 }
 
-func (r *VDOConfigReconciler) compareVersions(currentVersion, latestVersion string) (version string) {
+// compareVersions checks if the given version lies between min and max version
+func (r *VDOConfigReconciler) compareVersions(minVersion, currentVersion, maxVersion string) (bool, error) {
+	minVersionList := strings.Split(minVersion, ".")
```

Comment:
seems https://github.com/hashicorp/go-version would be a better fit here to compare versions